src/datastore/pg-write-store.ts

```diff
@@ -1963,13 +1965,33 @@ export class PgWriteStore extends PgStore {
     for (const txId of txIds) {
       logger.verbose(`Restoring mempool tx: ${txId}`);
     }
-    const updateResults = await sql<{ tx_id: string }[]>`
+
+    const updatedRows = await sql<{ tx_id: string }[]>`
       UPDATE mempool_txs
       SET pruned = false
       WHERE tx_id IN ${sql(txIds)}
       RETURNING tx_id
     `;
-    const restoredTxs = updateResults.map(r => r.tx_id);
+
+    const restoredTxs = updatedRows.map(r => r.tx_id);
+
+    // txs that didnt exist in the mempool need to be inserted into the mempool
+    if (updatedRows.length < txIds.length) {
+      const updatedTxs = updatedRows.map(r => r.tx_id);
+      const txsRequiringInsertion = txIds.filter(txId => !updatedTxs.includes(txId));
+
+      const txs: TxQueryResult[] = await sql`
+        SELECT ${sql(TX_COLUMNS)} FROM txs
+        WHERE tx_id IN ${sql(txsRequiringInsertion)}
```

Comment:
There can be multiple rows in `txs` for a given `tx_id`, primarily because of microblock re-orgs. I think this query needs something like
```sql
SELECT DISTINCT ON(tx_id) tx_id, ... 
ORDER BY tx_id, block_height DESC, microblock_sequence DESC, tx_index DESC, event_index DESC
```

That would ensure only 1 result per tx_id, and also add determinism to which tx is selected.

Reply:
Got it. I removed event_index because it's not in the TX_COLUMNS array, but I can easily add it if you think it's worth keeping.